worker/worker.go

```diff
@@ -0,0 +1,84 @@
+package worker
+
+import (
+	"encoding/json"
+	"log"
+
+	amqp "github.com/rabbitmq/amqp091-go"
+	"gorm.io/gorm"
+	"github.com/mini-maxit/worker/utils"
+)
+
+// Work starts the worker process
+func Work() {
+	
+	
+	db := connectToDatabase()
+
+	conn, ch := connectToRabbitMQ()
+	defer conn.Close()
+	defer ch.Close()
+
+	// Declare a queue
+	q, err := ch.QueueDeclare(
+		"worker_queue", // name
+		false,          // durable
+		false,          // delete when unused
+		false,          // exclusive
+		false,          // no-wait
+		nil,            // arguments
+	)
+	utils.FailOnError(err, "Failed to declare a queue")
+
+	// Consume messages from the queue
+	msgs, err := ch.Consume(
+		q.Name, // queue
+		"",     // consumer
```

Comment:
Shouldn't we specify the name of the consumer. if not, why?

Reply:
leaving it empty means that the rabbit mq will automatically generate a unique tag. As we do not need much control over the consumer I think it is not needed.

Reply:
agreed